Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java

```diff
@@ -389,7 +392,32 @@ public synchronized void clearSurface()
           NativeLibrary.WaitUntilDoneBooting();
         }
 
-        NativeLibrary.SurfaceDestroyed();
+        // AlertMessages while booting the core will dereference nullptr. Silently dismissing
+        // potential core issues is bad so toast and end emulation.
+        if (NativeLibrary.IsBooting())
+        {
+          if (NativeLibrary.GetAlertMessageText() != null)
+          {
+            String alertMessageError = "AlertMessage \"" + NativeLibrary.GetAlertMessageText() +
+                    "\" attempted to destroy surface while booting the core.";
+
+            Log.error("[EmulationFragment] " + alertMessageError);
+            Toast.makeText(context, alertMessageError, Toast.LENGTH_LONG).show();
+          }
+          else
+          {
+            String generalError = "Attempted to destroy surface while booting the core.";
+
+            Log.error("[EmulationFragment] " + generalError);
+            Toast.makeText(context, generalError, Toast.LENGTH_LONG).show();
+          }
+          stop();
```

Comment:
Are you sure that calling `StopEmulation` while booting works correctly? This seems very fishy to me.